client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-files/project-files.component.ts

```diff
@@ -34,52 +48,216 @@ export class ProjectFilesComponent {
+          }
+        },
+        error: (err) => {
+          this.snackbarService.open('Failed to load project attachments.', 'Close', {
+            duration: 5000,
+            panelClass: 'snackbar-error',
+          });
+        },
+      });
     }
-  ];
+  }
 
-  openFileUploadModal(){
+  openFileUploadModal() {
     const dialogRef = this.dialog.open(AddAttachmentComponent, {
       width: '1000px',
     });
 
     dialogRef.afterClosed().subscribe(result => {
       if (result?.file) {
-        this.uploadFile(result.file)
+        this.uploadFile(result.file);
+      }
+      if (result?.description) {
+        this.attachmentDescription = result.description;
       }
     })
   }
 
-  uploadFile(file: File): void{
-    this.projectService.uploadDocument({
-      file
-    }).subscribe({
+  uploadFile(file: File): void {
+    this.projectService.uploadDocument({ file }).subscribe({
       next: (response) => {
-        this.snackbarService.open(
-          this.messages.fileUploadSuccess,
-          'OK',
-          { duration: 5000, panelClass: 'snackbar-success' },
-        );
+        if (response) {
+          this.uploadAttachment(file, response);
+        }
       },
       error: () => {
+        this.snackbarService.open('The spatial file was not uploaded.', 'Close', {
+          duration: 5000,
+          panelClass: 'snackbar-error',
+        });
       }
-    })
+    });
   }
 
-  deleteFile(file:any){
-    //to do
+  uploadAttachment(file: File, response: any): void {
+    const fileExtension = file.name.split('.').pop()?.toLowerCase();
+
+    if (!fileExtension) {
+      this.snackbarService.open('The spatial file was not uploaded because the file format is not accepted.', 'Close', {
+        duration: 10000,
+        panelClass: ['snackbar-error'],
+      });
+      return;
+    }
+
+    const attachment: FileAttachment = {
+      sourceObjectNameCode: { sourceObjectNameCode: "PROJECT" },
+      sourceObjectUniqueId: this.projectGuid,
+      documentPath: file.name,
+      fileIdentifier: response.fileId,
+      attachmentContentTypeCode: { attachmentContentTypeCode: "DOCUMENT" },
+      attachmentDescription: this.attachmentDescription,
+      attachmentReadOnlyInd: false,
+    };
+
+    this.attachmentService.createProjectAttachment(this.projectGuid, attachment).subscribe({
+      next: (response) => {
+        if (response) {
+          this.uploadedBy = response?.uploadedByUserId;
+
+          this.spatialService.extractCoordinates(file).then(response => {
+            if (response) {
+              this.updateProjectBoundary(file, response)
+            }
+          })
+
+          this.loadProjectAttachments();
+        }
+      },
+      error: () => {
+        this.snackbarService.open('Failed to upload file. Please try again.', 'Close', {
+          duration: 10000,
+          panelClass: 'snackbar-error',
+        });
+      },
+    });
+  }
+
+  updateProjectBoundary(file: File, response: number[][]) {
```

Comment:
if we pass number[][] for the response, what do we do when the geometry file contains multi-polygons represent by number[][][].

Reply:
Updated to Position[][][]. Seems to work with both, with the serialization of the geojson being handled by JTSModule in the API.